view/theme/frio/style.php

```diff
@@ -1,40 +1,75 @@
 
-if(! $a->install) {
-	// Get the UID of the channel owner
+$schemecss = "";
+$schemecssfile = false;
+$scheme_modified = 0;
+
+if (! $a->install) {
+	// Get the UID of the profile owner.
 	$uid = get_theme_uid();
-	if($uid) {
-		load_pconfig($uid,'frio');
+	if ($uid) {
+		PConfig::load($uid, 'frio');
+
+		// Load the profile owners pconfig.
+		$schema           = PConfig::get($uid, "frio", "schema");
+		$nav_bg           = PConfig::get($uid, "frio", "nav_bg");
+		$nav_icon_color   = PConfig::get($uid, "frio", "nav_icon_color");
+		$link_color       = PConfig::get($uid, "frio", "link_color");
+		$bgcolor          = PConfig::get($uid, "frio", "background_color");
+		$contentbg_transp = PConfig::get($uid, "frio", "contentbg_transp");
+		$background_image = PConfig::get($uid, "frio", "background_image");
+		$bg_image_option  = PConfig::get($uid, "frio", "bg_image_option");
+		$modified         = PConfig::get($uid, "frio", "css_modified");
+
+		// There is maybe the case that the user did never modify the theme settings.
+		// In this case we store the present time.
+		if (empty($modified)) {
+			PConfig::set($uid, 'frio', 'css_modified', time());
+		}
+	} else {
+		Config::load('frio');
+
+		// Load frios system config.
+		$schema           = Config::get("frio", "schema");
+		$nav_bg           = Config::get("frio", "nav_bg");
+		$nav_icon_color   = Config::get("frio", "nav_icon_color");
+		$link_color       = Config::get("frio", "link_color");
+		$bgcolor          = Config::get("frio", "background_color");
+		$contentbg_transp = Config::get("frio", "contentbg_transp");
+		$background_image = Config::get("frio", "background_image");
+		$bg_image_option  = Config::get("frio", "bg_image_option");
+		$modified         = Config::get("frio", "css_modified");
+
+		// There is maybe the case that the user did never modify the theme settings.
+		// In this case we store the present time.
+		if (empty($modified)) {
+			Config::set('frio', 'css_modified', time());
+		}
 	}
-	// Load the owners pconfig
-	$schema		= get_pconfig($uid, "frio", "schema");
-	$nav_bg		= get_pconfig($uid, "frio", "nav_bg");
-	$nav_icon_color = get_pconfig($uid, "frio", "nav_icon_color");
-	$link_color	= get_pconfig($uid, "frio", "link_color");
-	$bgcolor	= get_pconfig($uid, "frio", "background_color");
-	$contentbg_transp = get_pconfig($uid, "frio", "contentbg_transp");
-	$background_image = get_pconfig($uid, "frio", "background_image");
-	$bg_image_option = get_pconfig($uid, "frio", "bg_image_option");
 }
 
 // Now load the scheme.  If a value is changed above, we'll keep the settings
 // If not, we'll keep those defined by the schema
 // Setting $schema to '' wasn't working for some reason, so we'll check it's
 // not --- like the mobile theme does instead.
-// Allow layouts to over-ride the schema
-if($_REQUEST['schema']) {
+// Allow layouts to over-ride the schema.
+if ($_REQUEST['schema']) {
 	$schema = $_REQUEST['schema'];
 }
+
+$schema = basename($schema);
```

Comment:
`$schema` isn’t always initialized at this point.

Reply:
This is a point.

Reply:
I can't understand if you're joking or not, `basename()` or any other function shouldn't be called on a potentially undeclared variable. As a general rule, it can produce unexpected behaviors, and with the wrong PHP config (`register_globals` turned on) it can become a security vulnerability.

Reply:
I just was confirming you that this should be done.